## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 423963`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16188155173


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.torchrl</summary>
<pre>INFO:wheel:adding 'torchrl/trainers/helpers/logger.py'
INFO:wheel:adding 'torchrl/trainers/helpers/losses.py'
INFO:wheel:adding 'torchrl/trainers/helpers/models.py'
INFO:wheel:adding 'torchrl/trainers/helpers/replay_buffer.py'
INFO:wheel:adding 'torchrl/trainers/helpers/trainers.py'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/licenses/LICENSE'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/METADATA'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/WHEEL'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/top_level.txt'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/RECORD'
INFO:root:removing build/bdist.linux-x86_64/wheel
Successfully built torchrl-0.8.1-cp312-cp312-linux_x86_64.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
buildPhase completed in 42 seconds
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for torchrl-0.8.1-cp312-cp312-linux_x86_64.whl
  - tensordict<0.9.0,>=0.8.3 not satisfied by version 0.9.0</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.torchrl</summary>
<pre>INFO:wheel:adding 'torchrl/trainers/helpers/logger.py'
INFO:wheel:adding 'torchrl/trainers/helpers/losses.py'
INFO:wheel:adding 'torchrl/trainers/helpers/models.py'
INFO:wheel:adding 'torchrl/trainers/helpers/replay_buffer.py'
INFO:wheel:adding 'torchrl/trainers/helpers/trainers.py'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/licenses/LICENSE'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/METADATA'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/WHEEL'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/top_level.txt'
INFO:wheel:adding 'torchrl-0.8.1.dist-info/RECORD'
INFO:root:removing build/bdist.linux-aarch64/wheel
Successfully built torchrl-0.8.1-cp312-cp312-linux_aarch64.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
buildPhase completed in 51 seconds
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for torchrl-0.8.1-cp312-cp312-linux_aarch64.whl
  - tensordict<0.9.0,>=0.8.3 not satisfied by version 0.9.0</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.tensordict</summary>
<pre>FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype0-device1] - RuntimeError: torch_shm_manager at "/nix/store/qbvkrgb3w50f3ah074y9070spcca07sm-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-None] - RuntimeError: torch_shm_manager at "/nix/store/qbvkrgb3w50f3ah074y9070spcca07sm-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-device1] - RuntimeError: torch_shm_manager at "/nix/store/qbvkrgb3w50f3ah074y9070spcca07sm-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_limitations - EOFError
= 23 failed, 26477 passed, 4865 skipped, 302 deselected, 73 warnings in 96.81s (0:01:36) =</pre>
</details>
<details>
<summary>python313Packages.tensordict</summary>
<pre>FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype0-device1] - RuntimeError: torch_shm_manager at "/nix/store/8qfb6jhw92ld4pxjnbi3g7wzny010m0q-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-None] - RuntimeError: torch_shm_manager at "/nix/store/8qfb6jhw92ld4pxjnbi3g7wzny010m0q-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-device1] - RuntimeError: torch_shm_manager at "/nix/store/8qfb6jhw92ld4pxjnbi3g7wzny010m0q-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_limitations - EOFError
= 23 failed, 26477 passed, 4865 skipped, 302 deselected, 73 warnings in 98.02s (0:01:38) =</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.tensordict</li>
    <li>python312Packages.tensordict.dist</li>
    <li>python313Packages.tensordict</li>
    <li>python313Packages.tensordict.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.tensordict</summary>
<pre>FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype0-device1] - RuntimeError: torch_shm_manager at "/nix/store/brw0knlmj41cxn822agj7mvlsl1m4va7-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-None] - RuntimeError: torch_shm_manager at "/nix/store/brw0knlmj41cxn822agj7mvlsl1m4va7-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-device1] - RuntimeError: torch_shm_manager at "/nix/store/brw0knlmj41cxn822agj7mvlsl1m4va7-python3.12-torch-2.7.1/lib/python3.12/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_limitations - EOFError
= 23 failed, 26477 passed, 4865 skipped, 302 deselected, 73 warnings in 48.79s =</pre>
</details>
<details>
<summary>python313Packages.tensordict</summary>
<pre>FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype0-device1] - RuntimeError: torch_shm_manager at "/nix/store/bjy6ax4pygbyd2p60lpc04yzpnimd91x-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-None] - RuntimeError: torch_shm_manager at "/nix/store/bjy6ax4pygbyd2p60lpc04yzpnimd91x-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestTensorDictRepr::test_repr_share_memory[dtype1-device1] - RuntimeError: torch_shm_manager at "/nix/store/bjy6ax4pygbyd2p60lpc04yzpnimd91x-python3.13-torch-2.7.1/lib/python3.13/site-packages/torch/bin/torch_shm_manager": Operation not permitted
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update_-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-inplace-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair0] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair1] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair2] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair3] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_memmap_single[update-shared-pair4] - EOFError
FAILED test/test_tensordict.py::TestNonTensorData::test_shared_limitations - EOFError
= 23 failed, 26477 passed, 4865 skipped, 302 deselected, 73 warnings in 51.41s =</pre>
</details>
</details>

--- a/pkgs/development/python-modules/gymnasium/default.nix
+++ b/pkgs/development/python-modules/gymnasium/default.nix
@@ -15,6 +15,10 @@
   pythonOlder,
   importlib-metadata,
 
+  # optional-dependencies
+  # atari
+  ale-py,
+
   # tests
   array-api-compat,
   dill,
@@ -53,6 +57,12 @@ buildPythonPackage rec {
     numpy
     typing-extensions
   ] ++ lib.optionals (pythonOlder "3.10") [ importlib-metadata ];
+
+  optional-dependencies = {
+    atari = [
+      ale-py
+    ];
+  };
 
   pythonImportsCheck = [ "gymnasium" ];
 

--- a/pkgs/development/python-modules/tensordict/default.nix
+++ b/pkgs/development/python-modules/tensordict/default.nix
@@ -28,14 +28,14 @@
 
 buildPythonPackage rec {
   pname = "tensordict";
-  version = "0.8.3";
+  version = "0.9.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "pytorch";
     repo = "tensordict";
     tag = "v${version}";
-    hash = "sha256-d/6JKGFcFLXY9pxsnP27uwnAnIQ9EKvfTS30DCwQrCM=";
+    hash = "sha256-actBFzWb2JBPsLhRZiD6zRpk7eyX2OHUPMU9JpJ90Wc=";
   };
 
   build-system = [

--- a/pkgs/development/python-modules/torchrl/default.nix
+++ b/pkgs/development/python-modules/torchrl/default.nix
@@ -17,12 +17,26 @@
   torch,
 
   # optional-dependencies
-  ale-py,
-  gym,
-  pygame,
-  torchsnapshot,
+  # atari
   gymnasium,
+  # checkpointing
+  torchsnapshot,
+  # gym-continuous
   mujoco,
+  # llm
+  accelerate,
+  datasets,
+  einops,
+  immutabledict,
+  langdetect,
+  nltk,
+  playwright,
+  protobuf,
+  safetensors,
+  sentencepiece,
+  transformers,
+  vllm,
+  # offline-data
   h5py,
   huggingface-hub,
   minari,
@@ -32,7 +46,9 @@
   scikit-learn,
   torchvision,
   tqdm,
+  # rendering
   moviepy,
+  # utils
   git,
   hydra-core,
   tensorboard,
@@ -48,14 +64,14 @@
 
 buildPythonPackage rec {
   pname = "torchrl";
-  version = "0.8.1";
+  version = "0.9.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "pytorch";
     repo = "rl";
     tag = "v${version}";
-    hash = "sha256-ANoqIAVKSq023hG83Q71t8oLzud1LeVN5WVPYL3nOks=";
+    hash = "sha256-afaWDX5lIAoGTfrBSqrktYoA1S4hv6ogBaKYHc8dQ6E=";
   };
 
   build-system = [
@@ -73,15 +89,25 @@ buildPythonPackage rec {
   ];
 
   optional-dependencies = {
-    atari = [
-      ale-py
-      gym
-      pygame
-    ];
+    atari = gymnasium.optional-dependencies.atari;
     checkpointing = [ torchsnapshot ];
     gym-continuous = [
       gymnasium
       mujoco
+    ];
+    llm = [
+      accelerate
+      datasets
+      einops
+      immutabledict
+      langdetect
+      nltk
+      playwright
+      protobuf
+      safetensors
+      sentencepiece
+      transformers
+      vllm
     ];
     offline-data = [
       h5py
@@ -131,10 +157,31 @@ buildPythonPackage rec {
     ]
     ++ optional-dependencies.atari
     ++ optional-dependencies.gym-continuous
+    ++ optional-dependencies.llm
     ++ optional-dependencies.rendering;
 
   disabledTests =
     [
+      # Require network
+      "test_create_or_load_dataset"
+      "test_from_text_env_tokenizer"
+      "test_from_text_env_tokenizer_catframes"
+      "test_from_text_rb_slicesampler"
+      "test_generate"
+      "test_get_dataloader"
+      "test_get_scores"
+      "test_preproc_data"
+      "test_prompt_tensordict_tokenizer"
+      "test_reward_model"
+      "test_tensordict_tokenizer"
+      "test_transform_compose"
+      "test_transform_model"
+      "test_transform_no_env"
+      "test_transform_rb"
+
+      # ray.exceptions.RuntimeEnvSetupError: Failed to set up runtime environment
+      "TestRayCollector"
+
       # torchrl is incompatible with gymnasium>=1.0
       # https://github.com/pytorch/rl/discussions/2483
       "test_resetting_strategies"
@@ -193,6 +240,16 @@ buildPythonPackage rec {
       # AssertionError: assert tensor([51.]) == ((5 * 11) + 2)
       "test_vecnorm_parallel_auto"
     ];
+
+  disabledTestPaths = [
+    # ERROR collecting test/smoke_test.py
+    # import file mismatch:
+    # imported module 'smoke_test' has this __file__ attribute:
+    #   /build/source/test/llm/smoke_test.py
+    # which is not the same as the test file we want to collect:
+    #   /build/source/test/smoke_test.py
+    "test/llm"
+  ];
 
   meta = {
     description = "Modular, primitive-first, python-first PyTorch library for Reinforcement Learning";

